libraries/SRV_Channel/SRV_Channel.h

```diff
@@ -183,7 +183,7 @@ class SRV_Channel {
 
     // check if a function is valid for indexing into functions
     static bool valid_function(Aux_servo_function_t fn) {
-        return fn >= 0 && fn < k_nr_aux_servo_functions;
+        return fn > k_none && fn < k_nr_aux_servo_functions;
```

Comment:
This looks like a drive by? I fear un-intended consequences. 

I think it means that setting servos to none at runtime will no longer stop there PWM output.

Reply:
where is the output currrently stopped?

Reply:
https://github.com/ArduPilot/ardupilot/blob/1521760a1713a4911956d6c6e79e296be751fd4e/libraries/SRV_Channel/SRV_Channels.cpp#L330-L332

I have not tested of course....

Maybe we could just special case none here:

https://github.com/ArduPilot/ardupilot/blob/1521760a1713a4911956d6c6e79e296be751fd4e/libraries/SRV_Channel/SRV_Channel_aux.cpp#L213-L224

Although the PWM channel would now not be enabled a few lines above. 

Which mask do you need to be correct?

Reply:
Motor enabled masks

Reply:
What about:
```
        if (channels[i].valid_function()) {
            channels[i].calc_pwm(functions[channels[i].function.get()].output_scaled);
        } else {
            channels[i].calc_pwm(0.0f);
        }
```
?

Reply:
Surely GPIO currently has the same issue?

Reply:
Actually I have a better fix that fixes a pre-existing issue

Reply:
... and the fix uncovers a bug in SITL